pytorch_lightning/utilities/memory.py

```diff
@@ -184,6 +184,6 @@ def get_model_size_mb(model: Module) -> float:
         Number of megabytes in the parameters of the input module.
     """
     model_size = _ByteCounter()
-    torch.save(model.state_dict(), model_size)
+    torch.save(model.state_dict(), cast(IO[bytes], model_size))
```

Comment:
Did you consider making `_ByteCounter` implement the `BytesIO` interface?

Reply:
Thanks for. the suggestion, I tried this but mypy complains about 
> pytorch_lightning/utilities/memory.py:35: error: Argument 1 of "write" is incompatible with supertype "BufferedIOBase"; supertype defines the argument type as "Union[bytes, Union[bytearray, memoryview, array[Any], mmap, _CData]]"  [override]
pytorch_lightning/utilities/memory.py:35: note: This violates the Liskov substitution principle"
```
class _ByteCounter(BytesIO):
    """Accumulate and stores the total bytes of an object."""

    def __init__(self) -> None:
        self.nbytes: int = 0

    def write(self, data: bytes) -> int:
        """Stores the total bytes of the data."""
        self.nbytes += len(data)
        return self.nbytes

    def flush(self) -> None:
        pass

```
It doesn't like the argument type becoming more specific(bytes v.s. the complex union).  Not sure what's the best way to address this issue (changing the input type of `write` here?) that's why I go with the castings.
Let me know how do you think! Thanks!

Reply:
@carmocca

Reply:
I pushed a commit where we use `BytesIO` directly instead of our own class. Let's see if CI agrees.